module/os/linux/zfs/zfs_vnops_os.c

```diff
@@ -1651,6 +1651,19 @@ zfs_getattr_fast(struct user_namespace *user_ns, struct inode *ip,
 	mutex_enter(&zp->z_lock);
 
 	zpl_generic_fillattr(user_ns, ip, sp);
+
+	sa_bulk_attr_t bulk[1];
+	uint64_t crtime[2];
+	int error, count = 0;
+	SA_ADD_BULK_ATTR(bulk, count, SA_ZPL_CRTIME(zfsvfs), NULL, &crtime, 16);
+	if ((error = sa_bulk_lookup(zp->z_sa_hdl, bulk, count)) != 0) {
+		mutex_exit(&zp->z_lock);
+		ZFS_EXIT(zfsvfs);
+		return (-error);
+	}
+	ZFS_TIME_DECODE(&sp->btime, crtime);
+	sp->result_mask |= STATX_BTIME;
```

Comment:
We'll want to also wrap this block in an `#ifdef STATX_BTIME` block (or full autoconf check) since this field isn't available in pre-4.11 kernels.  As mentioned in #8509 there's also some additional statx flags we could be setting to indicate some attributes, encryption, and compression. It's be nice to add them all, but that could certainly be done over multiple PRs.